packages/js-flagsmith-v1-provider/src/lib/flagsmith-v1.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  Context,
+  FeatureProvider,
+  FlagEvaluationOptions,
+  FlagTypeError,
+  FlagValueParseError,
+} from '@openfeature/openfeature-js';
+import * as flagsmith from 'flagsmith-nodejs';
+
+/**
+ * Additional custom data in Flagsmith is associated with the user (traits), and not passed directly to the flag evaluation.
+ * This is a very basic transformer function that defines Flagsmith traits based on the OpenFeature context.
+ * 
+ * We may want to generalize this concept, and provide a default for all providers.
+ */
+const DEFAULT_CONTEXT_TRANSFORMER = (context: Context): void => {
+  if (typeof context?.userId === 'string') {
+    Object.keys(context).forEach((key) => {
+      if (
+        key !== 'userId' &&
+        (typeof context[key] === 'boolean' ||
+          typeof context[key] === 'string' ||
+          typeof context[key] === 'number')
+      ) {
+        flagsmith.setTrait(context.userId as string, key, context[key]);
+      }
+    });
+  }
+};
```

Comment:
@beeme1mr this is a shockingly simply "ContextTransformer". Technically, for Flagsmith, "transoformer" is probably a misnomer, since it needn't return anything, but just set traits on the user.

I guess this strategy isn't a problem. I assume since these methods aren't asynchronous, there's no chatter involved in setting traits, and it's only adding properties to a locally cached user until some server synchronization occurs, perhaps during flag evaluation. Maybe @dabeeeenster can confirm.